src/org/etools/j1939_84/controllers/part01/Part01Step26Controller.java

```diff
@@ -300,47 +310,588 @@ protected void run() throws Throwable {
+                    // 6.1.26.18.a. For all MY2024+ engines, Warn each PG query where no response was received.
+                    addFailure("6.1.26.18.a - No response was received from "
+                            + module.getModuleName() + " for PG "
+                            + pg);
+                }
+                if (getEngineModelYear() >= 2022 && getEngineModelYear() <= 2023) {
+                    // 6.1.26.18.b. For MY2022-23 engines, Warn each PG query, where no response was received
+                    addWarning("6.1.26.18.b - No response was received from "
+                            + module.getModuleName() + " for PG "
+                            + pg);
+                }
+            } else {
+                ghgTracking100HrPacket.getSpns()
+                                      .forEach(spn -> {
+                                          // 6.1.26.18.c. Fail PG query where any bin value received is greater than
+                                          // FAFFh.
+                                          if (spn.getRawValue() >= 0xFAFFL) {
+                                              addFailure("6.1.26.18.c - Bin value received is greater than 0xFAFF(h)"
+                                                      + module.getModuleName() + " for " + spn);
+                                          }
+                                          // 6.1.26.18.d. Fail PG query where any index value received is greater than
+                                          // FAh.
+                                          if (spn.getId() == 12691 && spn.getValue() > 0xFAL) {
+                                              addFailure("6.1.26.16.c - PG query index received was " + "");
+                                          }
+                                          if (GHG_ACTIVE_GREEN_HOUSE_100_HR == spn.getId() && spn.getValue() > 0) {
+                                              // 6.1.26.18.g. Fail each active 100 hr array value that is greater than
+                                              // zero
+                                              addFailure("6.1.26.18.g - Active 100 hr array value received was greater than zero from  "
+                                                      + module.getModuleName());
+                                          }
+                                          // FIXME:
+                                          // 6.1.26.18.f. Fail each response where the set of labels received is not a
+                                          // subset of the set of labels received for the lifetime active technology
+                                          // response.
+                                          // if (!expectedLabels.contains(spn.getLabel()
+                                          // .substring(spn.getLabel().indexOf("Hour Active") + 5))) {
+                                          // addFailure("6.1.26.18.f - " + spn.getLabel());
```

Comment:
Be sure we call out to @battjt  that we aren't completing this particular requirement in this release.